### PR TITLE
Add Two Counter Metric in BigQuery Write Schema Transform

### PR DESCRIPTION
**Pull request by nickuncaged1201**
Adding two metrics, an input PCollection element count and one write failure count in BigQuery write schema transform. The metrics will be used for Syndeo monitoring integration.

### CODE REVIEW

**Comment by nickuncaged1201** [comment]
@pabloem 
@manavgarg

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java [hunk 1]
```diff
@@ -222,6 +226,32 @@ public void setBigQueryServices(BigQueryServices testBigQueryServices) {
...
+          Metrics.counter(
+              BigQueryStorageWriteApiPCollectionRowTupleTransform.class, "error-counter");
+
+      @ProcessElement
+      public void process(ProcessContext c) {
+        bqErrorCounter.inc();
+        c.output(c.element());
+      }
+    }
+
```
Comment (pabloem) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java:254]: I wonder if you can simplify these as a single DoFn that receives the counter name as parameter?

Also, `Metrics.counter` is a little bit slow due to particulars of its implementation. How about you do something like:

```
def process(self, element):
  self._elements_in_bundle += 1

def finish_bundle(self):
  self.bqErrorCounter.inc(self._elements_in_bundle)
  self._elements_in_bundle = 0
```

WDYT?

Comment (nickuncaged1201) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java:254]: I see makes sense. Thanks for pointing out!

**Comment by pabloem** [comment]
this seems reasonable. We may add more metrics to the elm/err counter, but we're good as-is for now, huh?

**Comment by pabloem** [comment]
LGTM. LMK if we should merge

**Comment by nickuncaged1201** [comment]
I'll probably change the name of each counter to be more descriptive of the IO type they are instrumented for. But that can wait till I add in Pub/Sub read metrics and I'll make the change. Think this is good for now. Feel free to merge it if you think It's okay! 

Thanks @pabloem for all the suggestion!

**Review by manavgarg — APPROVED** [review]

**Comment by pabloem** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by pabloem** [comment]
lgtm thanks @nickuncaged1201 !

**Pull request merged**